src/main/service/SubscriptionService.ts

```diff
@@ -141,12 +141,46 @@ export class SubscriptionService {
         };
     }
 
-    public async unsubscribe(subscriptionId: string, userId: string): Promise<object> {
-        return subscriptionRequests.unsubscribe(subscriptionId, userId);
+    public async unsubscribe(subscriptionId: string, userId: string, userRole: string): Promise<boolean> {
+        const response = await subscriptionRequests.unsubscribe(subscriptionId, userId);
+        if (response) {
+            return await this.configureListTypeAfterUnsubscribe(userId, userRole);
+        }
+        return false;
     }
 
-    public async bulkDeleteSubscriptions(subscriptionIds: string[], userId: string): Promise<object> {
-        return subscriptionRequests.bulkDeleteSubscriptions(subscriptionIds, userId);
+    public async bulkDeleteSubscriptions(
+        subscriptionIds: string[],
+        userId: string,
+        userRole: string
```

Comment:
I think this is userProvenance instead of role

Reply:
I have changed them all to userProvenance